### DISCUSSION TITLE
I want to know the difference between server private key and agent private key

### DISCUSSION THREAD

**Discussion by bonnlee2024** [shellhub-io/shellhub#4092] [Q&A] [answered]
When I install the shellhub-ssh-1, I can see the ssh_private_key,  and when I install the agent, I can see the shellhub.key. 

But I didn’t know where were the public keys.  
So i want to know the use of the private keys.  Is private key just an identity ID?

**Comment by otavio** [comment, marked as answer]
In the ShellHub architecture, both the server and agent use private keys for secure communication. The server's private key, which is vital for establishing an SSH gateway, authenticates the server to clients and agents, while its public key is shared for verification. Agents generate their private keys to authenticate to the server, sharing their public keys for mutual trust.
